modelzoo/SimpleMultiTask/train.py

```diff
@@ -0,0 +1,781 @@
+                checkpoint_dir=checkpoint_dir,
+                save_checkpoint_steps=save_ckp_steps,
+                summary_dir=checkpoint_dir,
+                save_summaries_steps=save_steps,
+                config=sess_config) as sess:
+            while not sess.should_stop():
+                sess.run([model.loss, model.train_op])
+    print('Training completed.')
+
+
+def eval(sess_config, input_hooks, model, test_init_op, test_steps, output_dir, checkpoint_dir):
+    hooks = []
+    hooks.extend(input_hooks)
+
+    scaffold = tf.train.Scaffold(
+        local_init_op=tf.group(tf.local_variables_initializer(), test_init_op))
+    session_creator = tf.train.ChiefSessionCreator(
+        scaffold=scaffold, checkpoint_dir=checkpoint_dir, config=sess_config)
+    writer = tf.summary.FileWriter(os.path.join(output_dir, 'eval'))
+    merged = tf.summary.merge_all()
+
+    with tf.train.MonitoredSession(session_creator=session_creator,
+                                   hooks=hooks) as sess:
+        for _in in range(1, test_steps + 1):
+            if (_in != test_steps):
+                sess.run([model.acc_op, model.auc_op])
+                if (_in % 1000 == 0):
+                    print(f'Evaluation complete:[{_in}/{test_steps}]')
+            else:
+                eval_acc, eval_auc, events = sess.run([model.acc_op, model.auc_op, merged])
+                writer.add_summary(events, _in)
+                print(f'Evaluation complete:[{_in}/{test_steps}]')
+                print(f'ACC = {eval_acc}\nAUC = {eval_auc}')
+
+def main(stock_tf, tf_config=None, server=None):
+    # check dataset and count data set size
+    print('Checking dataset...')
+    train_file = os.path.join(args.data_location, 'taobao_train_data')
+    test_file = os.path.join(args.data_location, 'taobao_test_data')
+    if not os.path.exists(args.data_location):
+        raise ValueError(f'[ERROR] data location: {args.data_location} does not exist. '
+                         'Please provide valid path')
+
+    if not os.path.exists(train_file) or not os.path.exists(test_file):
+        raise ValueError('[ERROR] taobao_train_data or taobao_test_data does not exist '
+                         'in the given data_location. Please provide valid path')
+
+    no_of_training_examples = sum(1 for _ in open(train_file))
+    no_of_test_examples = sum(1 for _ in open(test_file))
+
+    # set batch size, eporch & steps
+    batch_size = math.ceil(
+                  args.batch_size / args.micro_batch
+                  ) if args.micro_batch and not stock_tf else args.batch_size
+    if args.steps == 0:
+        no_epochs = 1
+        train_steps = math.ceil(
+            (float(no_epochs) * no_of_training_examples) / batch_size)
+    else:
+        no_epochs = math.ceil(
+            (float(batch_size) * args.steps) / no_of_training_examples)
+        train_steps = args.steps
+    test_steps = math.ceil(float(no_of_test_examples) / batch_size)
```

Comment:
Please be sure that by `python train.py`, the model will be trained for at least 3000 steps. Because the test script in DeepRec will run `python train.py` and collect the gsteps between 2000~3000. The script is under DeepRec/tests/model_benchmark.

Reply:
Why is the model being enforced to run 3000 steps regardless of the dataset size? 
If a benchmark requires the models to be run for 3000 steps for performance measurements, it is better to add `python train.py --steps 3000` into benchmark scripts instead of plain `python train.py` and enforcing all of the models to run 3000 steps.

Reply:
Please change the default steps by `no_epochs` here.

Reply:
Since ali is planning to increase the batch_size value in order to show the best performance, please just ignore the model accuracy for such a small dataset. By the way, please update the performance data and then we can push the PR to alibaba.
Thanks.